### ISSUE TITLE
Add plugin configuration

### ISSUE THREAD

**Issue by rrousselGit** [invertase/dart_custom_lint#4]
- [ ] changing the main location

**Comment by kuhnroyal** [comment]
I just migrated to `0.2.0` - this is a huge improvement, thanks a lot!
Can we have the `CustomLintConfigs` accessible in the `LintRule.run` method? Either via `context` or directly.
I am currently passing it in the constructor, works but I think this could be a general requirement.

**Comment by rrousselGit** [comment]
@kuhnroyal We could, but generally for performance that's not a good thing.

The parsing of configs should be done once before at the start of the analysis. `LintRule.run` should be as fast as possible

**Comment by kuhnroyal** [comment]
Yea, I am talking about the parsed instance that is also available in `PluginBase.getLintRules`. Could that not also be accessible later via context or the like?

**Comment by rrousselGit** [comment]
Yes, but you misunderstood what I said:

For the sake of performance, you should deal with configs within `getLintRules`. Such that the job is done only once instead of whenever any file is analyzed.

**Comment by kuhnroyal** [comment]
I see, so if I want to check wether another rule is enabled, you suggest to do it there?

**Comment by rrousselGit** [comment]
Why do you want to make such a check?

custom_lint already do not execute your lint if it's disabled

**Comment by kuhnroyal** [comment]
I have 2 similar rules with basically the same pretty complex, hierarchical checks and I don't want to run it twice or have the time to split it up. So I need to check if the 2nd rule is also active.

But everything is fine, I can pass it via constructor.
